src/wallet/wallet.cpp

```diff
@@ -2740,23 +2715,56 @@ MinedType GetGeneratedType(const uint256& tx, unsigned int vout)
     // Side/Split Stake Support
     else if (wallettx.vout.size() >= 3)
     {
-        // Split Stake
-        if (wallettx.vout[vout].scriptPubKey == wallettx.vout[1].scriptPubKey)
+        // If output 1 is mine and the pubkey (address) for the output is the same as
+        // output 1, it is a split stake return from my stake.
+        if (wallet->IsMine(wallettx.vout[1]) != ISMINE_NO
```

Comment:
Also wouldn't it be better if we could store the `wallet->IsMine(wallettx.vout[1])` output so that we can re-use it later for L2732?